dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/OneAgentMetadataEnricher.java

```diff
@@ -70,57 +70,110 @@ public OneAgentMetadataEnricher(Logger logger) {
 
       // make sure key and value are set to non-null, non-empty values
       if ((key == null || key.isEmpty()) || (value == null || value.isEmpty())) {
-        logger.warning(String.format("could not parse OneAgent metadata line ('%s')", line));
+        getLogger().warning(String.format("could not parse OneAgent metadata line ('%s')", line));
         continue;
       }
       entries.add(new AbstractMap.SimpleEntry<>(key, value));
     }
     return entries;
   }
 
-  private List<String> getMetadataFileContent() {
-    String indirectionBaseName = "dt_metadata_e617c525669e072eebe3d0f08212e8f2";
-    String secretFileName = null;
-    try (BufferedReader reader =
-        new BufferedReader(new FileReader(String.format("%s.properties", indirectionBaseName)))) {
-      String line = "";
-      // this whole while block is to make sure this function will still work even if the contents
-      // of the
-      // indirection file were to change in the future.
+  /**
+   * Get the file name of the file to which OneAgent persists its parameters.
+   *
+   * @param fileContents A {@link Reader} object pointing at the indirection file. Will be read
+   *     using a {@link BufferedReader}
```

Comment:
```suggestion
   * @param fileContents A {@link Reader} object pointing at the indirection file.
```
Implementation detail